--- a/net/minecraft/client/gui/GuiChat.java
+++ b/net/minecraft/client/gui/GuiChat.java
@@ -34,6 +34,8 @@
 import org.lwjgl.opengl.GL11;
 import tv.twitch.chat.ChatUserInfo;
 
+import com.mtbs3d.minecrift.provider.MCOpenVR;
+
 public class GuiChat extends GuiScreen implements GuiYesNoCallback
 {
     private static final Set supportedProtocols = Sets.newHashSet(new String[] {"http", "https"});
@@ -118,7 +120,7 @@
             this.playerNamesFound = false;
         }
 
-        if (keyCode == 1)
+        if (keyCode == 1) //esc
         {
             this.mc.displayGuiScreen((GuiScreen)null);
         }
@@ -154,8 +156,15 @@
                 this.submitChatMessage(var3);
             }
 
+            //VIVE DONT CLOSE ON RETURN
+            // this.mc.displayGuiScreen((GuiScreen)null);
+            if (MCOpenVR.keyboardShowing) {
+                this.inputField.setText("");
+            } else {
             this.mc.displayGuiScreen((GuiScreen)null);
         }
+            
+        }
     }
 
     public void submitChatMessage(String p_146403_1_)
@@ -184,7 +193,7 @@
                 var1 = -1;
             }
 
-            if (!isShiftKeyDown())
+            if (!isShiftKeyDown() || pressShiftFake)
             {
                 var1 *= 7;
             }
